input-explainer.md

```diff
@@ -176,16 +176,30 @@ Most applications will want to visually represent the input sources somehow. The
   * `'tracked-pointer'`: If the `gripTransform` is not `null` an application-appropriate controller model should be drawn using that transform. If appropriate for the experience, the a visualization of the target ray and a cursor as described in the `'gaze'` should also be drawn.
   * `'screen'`: In all cases the point of origin of the target ray is obvious and no visualization is needed.
 
+Determining what to render is important as well. In some cases it's appropriate for the application to render a contextually appropriate model (such as a racket in a tennis game), in which case the physical appearance of the input source doesn't matter much. Many other times, though, it's desirable for the application to display a device that matches what the user is holding, especially when relaying instructions about it's use. Finally there are times when it's best to not render anything at all, such as when the XR device uses a transparent display and the user can see their hands and/or any tracked devices without app assistance.
+
+ The `XRInputSource`'s `renderId` is used to determine what should be rendered in the cases when an application specific model isn't being used. 
+
+  - A `renderId` of `''` (empty string) indicates the input source should not be rendered. Cursors and pointing rays should still be rendered as described above if needed by the application. Input sources with a `targetRayMode` of `'gaze'` or `'screen'` MUST have a `renderId` of `''`. `'tracked-pointer'` input sources should typically only have a `renderId` of `''` when using displays that allow the user to see their environment and the physical input source.
+  - The `renderId` may be `'unknown'` if the type of input source cannot be reliably identified or the UA determines that the input source type must be masked for any reason.
+  - Inline sessions MUST only expose `renderId`s of `''` or `'unknown'`.
+  - Otherwise the `renderId` should be a lower-case string that describes the physical input source. It should not include and indication of the handedness of the input source, as that should be implied by the `handedness` attribute.
+    - If the input source is a tracked hand without any held controller, the `renderId` MUST be `hand`
+    - For most controllers this should ideally be of the format `<vendor>-<product-id>`. For example: `oculus-touch`. UAs should make an effort to align on the strings that are returned for any given device.
```

Comment:
The example `<product-id>` seems to be the token `touch` rather than any kind of prescriptive USB PID hex code.  Per this open format, the product ID could also be the token `065B` or the token `065B-0002` :smile:

Reply:
More generally, I'm concerned about this handwave:
> UAs should make an effort to align on the strings that are returned for any given device.

How do we actually envision UAs aligning here in advance?  Does the hardware/platform vendor just work with at least one UA (perhaps their own?) to set a precedent, with other UAs expected to tag along behind?  Do we operate a central registry of "vendor" tags to prevent collisions?  Or do we use reverse DNS here to prevent collisions... except is it the DNS of the hardware vendor or the UA that arbitrarily assigned a vendor/product ID to the hardware vendor's product?

Perhaps this is a discussion for the F2F.

Reply:
Definitely a topic for the F2F